psp/models/recent_history.py

```diff
@@ -124,20 +124,21 @@ def __init__(
         config: PvSiteModelConfig,
         *,
         pv_data_source: PvDataSource,
-        nwp_data_source: NwpDataSource | None,
+        nwp_data_source: list[NwpDataSource] | None,
         regressor: Regressor,
         random_state: np.random.RandomState | None = None,
-        use_nwp: bool = True,
-        nwp_variables: list[str] | None = None,
-        nwp_dropout: float = 0.0,
+        use_nwp: list[str] | None = None,
+        nwp_variables: list[list[str]] | None = None,
+        # For future NWPs may need to update the drop_out
+        nwp_dropout: list[float] = [0.0, 0.0],
```

Comment:
FYI you never want lists (or anything mutable) as default arguments in python. Use `None` and then set it to [0, 0] in the function. You'll find plenty of explanations on the internet for why that is, here is one: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
Noted!